File: 20-ValidParentheses/20-ValidParentheses.cpp
// Last updated: 7/16/2026, 4:39:22 PM
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for( char &ch : s ){
            if(ch == '(')
                st.push(')');
            else if (ch == '{')
                st.push('}');
                else if (ch == '[')
                    st.push(']');
                    else if (st.empty() || st.top() != ch)
                        return false;
                            else 
                                st.pop();
        }

        return st.empty();
    }
};

File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 8/26/2026, 10:04:05 AM
class Solution:
    def isValid(self, s: str) -> bool:
        l = ['z']

        for i in s:
            if i == '(' or i == '{'  or i == '[':
                l.append(i)
            elif i == ')':
                if l[-1] == '(':
                    l.pop()
                else: 
                    return False
            elif i == ']':
                if l[-1] == '[':
                    l.pop()
                else: 
                    return False

            elif i == '}':
                if l[-1] == '{':
                    l.pop()
                else: 
                    return False

        if len(l) == 1:
            return True
        return False

        